tests/unit/test_opensslwrapper.cpp

```diff
@@ -340,3 +340,41 @@ TEST_F(OpenSSLWrapperTest, testEngineFinishThrowsException)
     EXPECT_CALL(_mock(), SSL_ENGINE_finish(engine)).WillOnce(Return(0));
     EXPECT_THROW(_ENGINE_finish(engine), OpenSSLException);
 }
+
+TEST_F(OpenSSLWrapperTest, testEngineCtrlCmd)
```

Comment:
I don't really think that we need tests for the wrap layer. At least we usually don't have this if the function just implements error handling using the CallChecked templates.

Reply:
Agreed but I guess these don't hurt much as well